homeassistant/components/media_player/braviatv.py

```diff
@@ -0,0 +1,473 @@
+"""
+Support for interface with a Sony Bravia TV.
+
+By Antonio Parraga Navarro
+
+dedicated to Isabel
+
+"""
+import logging
+import os
+import json
+import re
+from io import StringIO
+from homeassistant.loader import get_component
+from homeassistant.components.media_player import (
+    SUPPORT_NEXT_TRACK, SUPPORT_PAUSE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_TURN_OFF, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_STEP,
+    SUPPORT_VOLUME_SET, SUPPORT_SELECT_SOURCE, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF, STATE_ON)
+
+REQUIREMENTS = [
+    'https://github.com/aparraga/braviarc/archive/0.2.1.zip'
+    '#braviarc==0.2.1']
+
+BRAVIA_CONFIG_FILE = 'bravia.conf'
+CLIENTID_PREFIX = 'HomeAssistant'
+NICKNAME = 'Home Assistant'
+
+# Map ip to request id for configuring
+_CONFIGURING = {}
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_BRAVIA = SUPPORT_PAUSE | SUPPORT_VOLUME_STEP | \
+                 SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_SET | \
+                 SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
+                 SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+
+def _jdata_build(method, params):
+    if params:
+        ret = json.dumps({"method": method,
+                          "params": [params],
+                          "id": 1,
+                          "version": "1.0"})
+    else:
+        ret = json.dumps({"method": method,
+                          "params": [],
+                          "id": 1,
+                          "version": "1.0"})
+    return ret
+
+
+def _config_from_file(filename, config=None):
+    """Small configuration file management function."""
+    if config:
+        # We're writing configuration
+        bravia_config = _config_from_file(filename)
+        if bravia_config is None:
+            bravia_config = {}
+        new_config = bravia_config.copy()
+        new_config.update(config)
+        try:
+            with open(filename, 'w') as fdesc:
+                string_io = StringIO()
+                json.dump(new_config, string_io)
```

Comment:
what? You can either pass in `fdesc` straight into `json.dump` or you call `json.dumps(…)` and get the string value which you pass to `fdesc.write(…)`. There is no need for `StringIO`.